### PR TITLE
fix: cleanup bugs from PR to move envoy proxy image

### PR DESCRIPTION
**Pull request by jaellio**
**Description**:

Removes a typo from a comment and adds a default value for the
envoy image in the default mesh config.

**Affected area**:

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ X ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Demo                   [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
no

### CODE REVIEW

File: cmd/init-osm-controller/init-osm-controller.go [hunk 1]
```diff
@@ -65,6 +65,7 @@ func createDefaultMeshConfig() *v1alpha1.MeshConfig {
 		Spec: v1alpha1.MeshConfigSpec{
 			Sidecar: v1alpha1.SidecarSpec{
 				LogLevel:                      "error",
+				EnvoyImage:                    "envoyproxy/envoy-alpine:v1.17.2",
```
Comment (allenlsy) [inline cmd/init-osm-controller/init-osm-controller.go:68]: Curious, without this, what Envoy image was installed previously?

**Review by allenlsy — APPROVED** [review]

**Review by snehachhabria — APPROVED** [review]

File: pkg/configurator/validating_webhook.go [hunk 2]
```diff
@@ -319,7 +319,7 @@ func checkEnvoyLogLevels(configMapField, configMapValue string) bool {
 	return valid
 }
 
-// checkEnvoyLogLevels checks that the field value is a valid log level
+// checkEnvoyImage checks that the namespace and registry of the envoy proxy sidecar image are valid
```
Comment (shashankram) [inline pkg/configurator/validating_webhook.go:322]: What does namespace imply?

Comment (jaellio) [reply pkg/configurator/validating_webhook.go]: Updated to `name`. I am not sure namespace was the correct term

Comment (shashankram) [reply pkg/configurator/validating_webhook.go]: Thanks, makes sense now.

**Pull request merged**